packages/components/src/menu/stories/index.story.tsx

```diff
@@ -64,300 +64,337 @@ const meta: Meta< typeof Menu > = {
 			source: { excludeDecorators: true },
 		},
 	},
-};
+} satisfies Meta< typeof Menu >;
 export default meta;
 
-export const Default: StoryFn< typeof Menu > = ( props: MenuProps ) => (
-	<Menu { ...props }>
-		<Menu.TriggerButton
-			render={ <Button __next40pxDefaultSize variant="secondary" /> }
-		>
-			Open menu
-		</Menu.TriggerButton>
-		<Menu.Popover>
-			<Menu.Item>
-				<Menu.ItemLabel>Label</Menu.ItemLabel>
-			</Menu.Item>
-			<Menu.Item>
-				<Menu.ItemLabel>Label</Menu.ItemLabel>
-				<Menu.ItemHelpText>Help text</Menu.ItemHelpText>
-			</Menu.Item>
-			<Menu.Item>
-				<Menu.ItemLabel>Label</Menu.ItemLabel>
-				<Menu.ItemHelpText>
-					The menu item help text is automatically truncated when
-					there are more than two lines of text
-				</Menu.ItemHelpText>
-			</Menu.Item>
-			<Menu.Item hideOnClick={ false }>
-				<Menu.ItemLabel>Label</Menu.ItemLabel>
-				<Menu.ItemHelpText>
-					This item doesn&apos;t close the menu on click
-				</Menu.ItemHelpText>
-			</Menu.Item>
-			<Menu.Item disabled>Disabled item</Menu.Item>
-			<Menu.Separator />
-			<Menu.Group>
-				<Menu.GroupLabel>Group label</Menu.GroupLabel>
-				<Menu.Item prefix={ <Icon icon={ customLink } size={ 24 } /> }>
-					<Menu.ItemLabel>With prefix</Menu.ItemLabel>
-				</Menu.Item>
-				<Menu.Item suffix="⌘S">With suffix</Menu.Item>
-				<Menu.Item
-					disabled
-					prefix={ <Icon icon={ formatCapitalize } size={ 24 } /> }
-					suffix="⌥⌘T"
-				>
-					<Menu.ItemLabel>
-						Disabled with prefix and suffix
-					</Menu.ItemLabel>
-					<Menu.ItemHelpText>And help text</Menu.ItemHelpText>
-				</Menu.Item>
-			</Menu.Group>
-		</Menu.Popover>
-	</Menu>
-);
-Default.args = {};
+type Story = StoryObj< MenuProps >;
 
-export const WithSubmenu: StoryFn< typeof Menu > = ( props: MenuProps ) => (
-	<Menu { ...props }>
-		<Menu.TriggerButton
-			render={ <Button __next40pxDefaultSize variant="secondary" /> }
-		>
-			Open menu
-		</Menu.TriggerButton>
-		<Menu.Popover>
-			<Menu.Item>Level 1 item</Menu.Item>
-			<Menu>
-				<Menu.SubmenuTriggerItem suffix="Suffix">
-					<Menu.ItemLabel>
-						Submenu trigger item with a long label
-					</Menu.ItemLabel>
-				</Menu.SubmenuTriggerItem>
+export const Default: Story = {
+	args: {
+		children: (
```

Comment:
The `csf-2-to-3` codemod would use `render: () => { ... }` instead of `args.children`. What's the difference? Seems like it allows you to specify the props type inline when passing it to `render`:

```
export const Default: StoryObj< typeof Menu > = {
	render: ( props: MenuProps ) => (
		<Menu { ...props }>
			/// ...
		</Menu>
	),

	args: {},
};
```

One difference to note is that the codemod uses `render:` but also specifically adds this to all stories that inherit from `Default`:

```
	args: {
		...Default.args,
	},
```

Reply:
When I was working on some of the TypeScript compilation hotspots, I noticed that not having a `render` function is better for performance, since it can bypass a lot of work. So we should avoid `render` and prefer `children` when possible.

Reply:
Agree with Lena, it seems more performant to use `children` when possible

Reply:
This makes sense to me 👍 

However, how do we maintain it as other people work on stories and introduce CSF3 ones? Should we add some docs or introduce linting? Any better ideas?

Reply:
We could document it, and we usually point it out in code review, but that's about as far as I think it needs to be "enforced". No biggie.

This was already a thing in CSF2, where some people would write completely new templates instead of just changing `args.children`.

Reply:
From what I know, using args over custom "render" function has always been a best practice, even prior to CSF3.

In case it helps, Storybook docs also point out that using args is preferred: https://storybook.js.org/docs/8.5/writing-stories#working-with-react-hooks:~:text=We%20recommend%20args%20as%20much%20as%20possible%20when%20writing%20your%20own%20stories.

Not sure if we should spend time writing docs or lining Storybook files?